### PR TITLE
Ros1 fixes heart anomaly detection

### PR DESCRIPTION
**Pull request by minhquoc0712**
Fixed:

- Update device code selection.
- Update argparse.
- Update input and output topic name, and docstring accordingly.

### CODE REVIEW

File: projects/opendr_ws/src/perception/scripts/heart_anomaly_detection.py [hunk 1]
```diff
@@ -25,11 +25,12 @@
...
+    def __init__(self, input_ecg_topic="/ecg/ecg", output_heart_anomaly_topic="/opendr/heart_anomaly",
+                 device="cuda", model="anbof"):
         """
         Creates a ROS Node for heart anomaly (atrial fibrillation) detection from ecg data
-        :param input_topic: Topic from which we are reading the input array data
-        :type input_topic: str
+        :param input_ecg_topic: Topic from which we are reading the input array data
+        :type input_ecg_topic: str
         :param prediction_topic: Topic to which we are publishing the predicted class
         :type prediction_topic: str
```
Comment (tsampazk) [inline projects/opendr_ws/src/perception/scripts/heart_anomaly_detection.py:35]: Mismatched docstring.

```suggestion
        :param output_heart_anomaly_topic: Topic to which we are publishing the predicted class
        :type output_heart_anomaly_topic: str
```

File: projects/opendr_ws/src/perception/scripts/heart_anomaly_detection.py [hunk 2]
```diff
@@ -83,17 +84,32 @@ def callback(self, msg_data):
...
+                        help="listen to input ECG data on this topic")
+    parser.add_argument("--model", type=str, default="anbof", help="model to be used for prediction: anbof or gru")
+    parser.add_argument("--output_heart_anomaly_topic", type=str, default="/opendr/heart_anomaly",
+                        help="Topic name for heart anomaly detection topic")
+    parser.add_argument("--device", type=str, default="cuda", help="Device to use (cpu, cuda)",
+                        choices=["cuda", "cpu"])
+
+    args = parser.parse_args()
+
     # Select the device for running
```
Comment (tsampazk) [inline projects/opendr_ws/src/perception/scripts/heart_anomaly_detection.py:100]: This comment could be removed
```suggestion
```

**Review by tsampazk — CHANGES_REQUESTED** [review]
Thank you for the updates in this node, changes look good and the node seems to run smoothly. I have left two very minor comments, and after those it will be good to go!

File: projects/opendr_ws/src/perception/scripts/heart_anomaly_detection.py [hunk 3]
```diff
@@ -83,17 +84,31 @@ def callback(self, msg_data):
...
-    except:
-        device = 'cpu'
 
+if __name__ == '__main__':
     parser = argparse.ArgumentParser()
-    parser.add_argument('input_topic', type=str, help='listen to input data on this topic')
-    parser.add_argument('model', type=str, help='model to be used for prediction: anbof or gru')
+    parser.add_argument("--input_ecg_topic", type=str, default="/ecg/ecg",
+                        help="listen to input ECG data on this topic")
+    parser.add_argument("--model", type=str, default="anbof", help="model to be used for prediction: anbof or gru")
```
Comment (ad-daniel) [inline projects/opendr_ws/src/perception/scripts/heart_anomaly_detection.py:92]:
```suggestion
    parser.add_argument("--model", type=str, default="anbof", help="model to be used for prediction: anbof or gru",
                        choices=["anbof", "gru"])
```

**Review by ad-daniel — CHANGES_REQUESTED** [review]
Just a detail, thank you

**Review by ad-daniel — APPROVED** [review]
Thank you

**Review by tsampazk — APPROVED** [review]

**Pull request merged**